### PR TITLE
Ledger app validation

### PR DESCRIPTION
**Pull request by kaladinlight**
Validate current app before device calls when specific open app is required

### CODE REVIEW

File: packages/hdwallet-ledger/src/ledger.ts [hunk 1]
```diff
@@ -322,25 +322,25 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
...
-   * Throws WrongApp error if symbol does not match app
-   * @param symbol - i.e. 'BCH'
-   * @returns {Promise<void>}
+   * Validate if a specific app is open
+   * Throws WrongApp error if app associated with coin is not open
+   * @param coin  Name of coin for app name lookup
    */
-  public async validateCurrentApp (symbol: string): Promise<void> {
-    if (!symbol) return
+  public async validateCurrentApp (coin: string): Promise<void> {
```
Comment (keepkeyjon) [inline packages/hdwallet-ledger/src/ledger.ts]:
```suggestion
  public async validateCurrentApp (coin: Coin): Promise<void> {
```

File: packages/hdwallet-ledger/src/ledger.ts [hunk 2]
```diff
@@ -322,25 +322,25 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
...
-  public async validateCurrentApp (symbol: string): Promise<void> {
-    if (!symbol) return
+  public async validateCurrentApp (coin: Coin): Promise<void> {
+    if (!coin) return
 
-    const expectedApp = appsUtil[symbol.toUpperCase()]
-    if (!expectedApp) {
-      throw new Error(`Cannot find app for symbol: ${symbol}`)
+    const { appName = undefined } = networksUtil[core.slip44ByCoin(coin)] || {}
+    console.log('appName', appName)
```
Comment (ryankeepkey) [inline packages/hdwallet-ledger/src/ledger.ts]: don't need this anymore

File: packages/hdwallet-ledger/src/ledger.ts [hunk 3]
```diff
@@ -322,25 +322,24 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
...
    */
-  public async validateCurrentApp (symbol: string): Promise<void> {
-    if (!symbol) return
+  public async validateCurrentApp (coin: core.Coin): Promise<void> {
+    if (!coin) return
 
-    const expectedApp = appsUtil[symbol.toUpperCase()]
-    if (!expectedApp) {
-      throw new Error(`Cannot find app for symbol: ${symbol}`)
+    const { appName = undefined } = networksUtil[core.slip44ByCoin(coin)] || {}
```
Comment (dviramontes) [inline packages/hdwallet-ledger/src/ledger.ts]: the `... || {}` part of this assignment makes me think - we would never get `undefined` as default.

File: packages/hdwallet-ledger/src/ledger.ts [hunk 4]
```diff
@@ -322,25 +322,24 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
...
    */
-  public async validateCurrentApp (symbol: string): Promise<void> {
-    if (!symbol) return
+  public async validateCurrentApp (coin: core.Coin): Promise<void> {
+    if (!coin) return
 
-    const expectedApp = appsUtil[symbol.toUpperCase()]
-    if (!expectedApp) {
-      throw new Error(`Cannot find app for symbol: ${symbol}`)
+    const appName = get(networksUtil[core.slip44ByCoin(coin)], 'appName', undefined)
```
Comment (keepkeyjon) [inline packages/hdwallet-ledger/src/ledger.ts]:
```suggestion
    const appName = get(networksUtil[core.slip44ByCoin(coin)], 'appName')
```

File: packages/hdwallet-ledger/src/ledger.ts (hunk 3 above)
Comment (kaladinlight) [reply packages/hdwallet-ledger/src/ledger.ts]: Updated with lodash get

**Pull request merged**